modules/services/davmail.nix

```diff
@@ -0,0 +1,142 @@
+{ config, lib, pkgs, ... }:
+let
+
+  inherit (lib)
+    attrNames boolToString concatMap concatStringsSep isAttrs isBool
+    mapAttrsRecursive mkDefault mkEnableOption mkIf mkOption optionalAttrs
+    types;
+
+  cfg = config.services.davmail;
+
+  settingsType = with types;
+    oneOf [ (attrsOf settingsType) str int bool ] // {
+      description =
+        "DavMail settings type (str, int, bool or attribute set thereof)";
+    };
+
+  toStr = val: if isBool val then boolToString val else toString val;
+
+  linesForAttrs = attrs:
+    concatMap (name:
+      let value = attrs.${name};
+      in if isAttrs value then
+        map (line: name + "." + line) (linesForAttrs value)
+      else
+        [ "${name}=${toStr value}" ]) (attrNames attrs);
+
+  settingsFile = pkgs.writeText "davmail.properties"
+    (concatStringsSep "\n" (linesForAttrs cfg.settings));
+
+in {
+
+  meta.maintainers = [ lib.hm.maintainers.bmrips ];
+
+  options.services.davmail = {
+
+    enable = mkEnableOption "DavMail, an MS Exchange gateway.";
+
+    imitateOutlook = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether DavMail pretends to be Outlook.";
+      example = true;
+    };
+
+    settings = mkOption {
+      type = settingsType;
```

Comment:
```suggestion
      type = types.attrsOf types.anything;
```

Reply:
This might be simpler but enables invalid values. Hence, I prefer to keep the more complex but accurate type.